lib/minheap.js

```diff
@@ -38,26 +51,46 @@ class MinHeap {
   }
 
   // This method returns true if the heap is empty
-  // Time complexity: ?
-  // Space complexity: ?
+  // Time complexity: O(1)
+  // Space complexity: O(1)
   isEmpty() {
-    throw new Error("Method not implemented yet...");
+    return this.store.length == 0;
   }
 
   // This helper method takes an index and
   //  moves it up the heap, if it is less than it's parent node.
   //  It could be **very** helpful for the add method.
-  // Time complexity: ?
-  // Space complexity: ?
+  // Time complexity: O(log n)
+  // Space complexity: O(log n)
   heapUp(index) {
-    throw new Error("Method not implemented yet...");
+    let parent = Math.floor((index - 1)/2);
+    if (parent < 1) parent = 0;
+
+    if (this.store[parent].key > this.store[index].key) {
+      this.swap(parent, index);
+      return this.heapUp(parent);
+    }
   }
 
   // This helper method takes an index and 
   //  moves it up the heap if it's smaller
   //  than it's parent node.
   heapDown(index) {
```

Comment:
This method works in _most_ cases, but I think you're missing an edge-case here.